tests/appsec/test_suspicious_attacker_blocking.py

```diff
@@ -0,0 +1,119 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+
+from utils import features
+from utils import interfaces
+from utils import remote_config as rc
+from utils import scenarios
+from utils import weblog
+
+
+CONFIG_ENABLED = (
+    "datadog/2/ASM_FEATURES/asm_features_activation/config",
+    {"asm": {"enabled": True}},
+)
+BLOCK_405 = (
+    "datadog/2/ASM/actions/config",
+    {"actions": [{"id": "block_custom", "parameters": {"status_code": 405, "type": "auto"}, "type": "block_request",}]},
+)
+
+EXCLUSIONS = (
+    "datadog/2/ASM/exclusions/config",
+    {
+        "exclusions": [
+            {
+                "id": "exc-000-001",
+                "on_match": "block_custom",
+                "conditions": [
+                    {
+                        "operator": "ip_match",
+                        "parameters": {"data": "suspicious_ips_data_id", "inputs": [{"address": "http.client_ip"}],},
+                    }
+                ],
+            }
+        ]
+    },
+)
+
+EXCLUSION_DATA = (
+    "datadog/2/ASM/exclusions_data/config",
+    {
+        "exclusion_data": [
+            {"id": "suspicious_ips_data_id", "type": "ip_with_expiration", "data": [{"value": "34.65.27.85"}],}
+        ]
+    },
+)
+
+BLOCK_416 = (
+    "datadog/2/ASM/actions/config",
+    {"actions": [{"id": "block_custom", "parameters": {"status_code": 416, "type": "auto"}, "type": "block_request",}]},
+)
+
+HEADERS_ATTACKER = {
+    "User-Agent": "dd-test-scanner-log-block",
+    "X-Real-Ip": "34.65.27.85",
+}
+HEADERS_REGULAR = {"User-Agent": "dd-test-scanner-log-block"}
+
+
+@scenarios.appsec_runtime_activation
+@features.suspicious_attacker_blocking
+class Test_Suspicious_Attacker_Blocking:
+    """A library should block requests after AppSec is activated via remote config,
+    using the blocking actions defined in the remote config.
+    """
+
+    def setup_block_suspicious_attacker(self):
+        self.config_state_1 = rc.rc_state.reset().apply()
+        self.response_1 = weblog.get("/waf/", headers=HEADERS_ATTACKER)
+
+        self.config_state_2 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.response_2 = weblog.get("/waf/", headers=HEADERS_ATTACKER)
+
+        self.config_state_3 = (
+            rc.rc_state.set_config(*BLOCK_405).set_config(*EXCLUSIONS).set_config(*EXCLUSION_DATA).apply()
+        )
+        self.response_3 = weblog.get("/waf/", headers=HEADERS_ATTACKER)
+        self.response_3b = weblog.get("/waf/", headers=HEADERS_REGULAR)
```

Comment:
It would be interesting to add another variant where the header is `dd-test-scanner-log` so that we can see that a non-blocking rule is converted to blocking.

Reply:
I added requests with non blocking rule to check if it's now blocking as expected (and it works with the python tracer).